### PR TITLE
Add ShowLatestVersions command handler

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: .atomist/tslint.json [hunk 1]
```diff
@@ -5,10 +5,14 @@
...
-        "interface-name": [true, "never-prefix"],
+        "interface-name": [
+            true,
+            "never-prefix"
+        ],
         "max-classes-per-file": false,
         "object-literal-sort-keys": false,
-        "no-console": false
+        "no-console": false,
+        "radix": false
```
Comment (ddgenome) [inline .atomist/tslint.json]: Do we really want to disable this rule?
https://palantir.github.io/tslint/rules/radix/

File: .atomist/tslint.json [hunk 2]
```diff
@@ -5,10 +5,14 @@
...
     },
     "rulesDirectory": []
-}
+}
```
Comment (ddgenome) [inline .atomist/tslint.json]: Please restore newline at end of file.

File: .atomist/handlers/command/ShowLatestVersions.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        const plan = new CommandPlan();
+        plan.add(npmGet);
+        return plan;
+    }
+}
+
+@ResponseHandler("ReceiveRugCliResponse", "Receive the pom.xml for the latest Rug CLI release")
+class ReceiveRugCliResponse implements HandleResponse<string> {
+
+    @Parameter({ description: "Rug version from JFrog", pattern: "^.*$" })
```
Comment (ddgenome) [inline .atomist/handlers/command/ShowLatestVersions.ts]: You could use `Pattern.semantic_version` here.

File: .atomist/handlers/command/ShowLatestVersions.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    }
+}
+
+@ResponseHandler("ReceiveNpmResponse", "Receive JSON for the latest @atomist/rugs release")
+class ReceiveNpmResponse implements HandleResponse<any> {
+
+    @Parameter({ description: "Rug version from JFrog", pattern: "^.*$" })
+    public rugVersion: string;
+
+    @Parameter({ description: "Rug CLI version from JFrog", pattern: "^.*$" })
```
Comment (ddgenome) [inline .atomist/handlers/command/ShowLatestVersions.ts]: Pattern.semantic_version

File: .atomist/handlers/command/ShowLatestVersions.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        const plan = new CommandPlan();
+        plan.add(npmGet);
+        return plan;
+    }
+}
+
+@ResponseHandler("ReceiveNpmResponse", "Receive JSON for the latest @atomist/rugs release")
+class ReceiveNpmResponse implements HandleResponse<any> {
+
+    @Parameter({ description: "Rug version from JFrog", pattern: "^.*$" })
```
Comment (ddgenome) [inline .atomist/handlers/command/ShowLatestVersions.ts]: Pattern.semantic_version

File: .atomist/handlers/command/ShowLatestVersions.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+/**
+ * Helper function to increment the major digit of a version string
+ * @param version version to increment
+ */
+function revVersion(version: string): string {
+    const parts = version.split(".", 2);
+    console.log(parts[0]);
+    const major = parseInt(parts[0]);
```
Comment (ddgenome) [inline .atomist/handlers/command/ShowLatestVersions.ts]: `parseInt(parts[0], 10)`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

File: .atomist/util/HttpClient.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            name: "http",
+            kind: "execute",
+            parameters: {
+                method,
+                url,
+                config: {
+                    headers,
+                    body,
+                },
+            },
```
Comment (ddgenome) [inline .atomist/util/HttpClient.ts]: Probably good to provide a generic `onError` here or use `handleErrors` from `@atomist/rugs/operations/CommonHandlers`.

File: README.md [hunk 8]
```diff
@@ -477,6 +477,42 @@ $ ( cd .atomist && yarn )
 
 [yarn]: https://yarnpkg.com/
 
+## ShowLatestVersions
```
Comment (ddgenome) [inline README.md]: Prior to this addition, the Rugs were in alphabetical order.

File: .atomist/handlers/command/ShowLatestVersions.ts [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+const rugUrl = "https://atomist.jfrog.io/atomist/rugs/com/atomist/rug/[RELEASE]/rug-[RELEASE].pom";
+const rugCliUrl = "https://atomist.jfrog.io/atomist/rugs/com/atomist/rug-cli/[RELEASE]/rug-cli-[RELEASE].pom";
+// npm uses a bit of a weired syntax to query scoped packages
+const rugsUrl = "https://registry.npmjs.org/@atomist%2Frugs/";
+
+/**
+ * Shows the latest versions for rug and rugs.
+ */
+@CommandHandler("ShowLatestVersions", "Shows the latest versions for rug and rugs")
```
Comment (ddgenome) [inline .atomist/handlers/command/ShowLatestVersions.ts]: "for rug, @atomist/rugs, and the Rug CLI"

**Comment by cdupuis** [comment]
@ddgenome yeah, the nesting is required as we currently don't have any support for _collecting_ responses from multiple `execute` instructions in one go. I think that is something we need to provide sooner or later.

**Comment by ddgenome** [comment]
Thanks for the updates!

**Pull request merged**